test/client/fixtures/runner/api/press-test.js

```diff
@@ -304,82 +322,108 @@ $(document).ready(function () {
     });
 
     asyncTest('B253200 - TestCafe doesn\'t emulate browsers behavior for press "enter" key on the focused HyperLink editor (link with href)', function () {
-        var iFrameSrc = window.QUnitGlobals.getResourceUrl('../../../data/runner/iframe.html', 'runner-iframe.html');
-        var linkHref  = window.QUnitGlobals.getResourceUrl('../../../data/focus-blur-change/iframe.html', 'focus-iframe.html');
-        var link      = $('<a>Link</a>').attr('href', linkHref).addClass(TEST_ELEMENT_CLASS)[0];
-        var iframe    = createIFrame(iFrameSrc);
-        var clicked   = false;
+        var iFrameSrc    = window.QUnitGlobals.getResourceUrl('../../../data/runner/iframe.html', 'runner-iframe.html');
+        var linkHref     = window.QUnitGlobals.getResourceUrl('../../../data/focus-blur-change/iframe.html', 'focus-iframe.html');
+        var link         = $('<a>Link</a>').attr('href', linkHref).addClass(TEST_ELEMENT_CLASS)[0];
+        var iframe       = createIFrame(iFrameSrc);
+        var clicked      = false;
+        var testFinished = false;
 
         link.onclick = function () {
             clicked = true;
         };
 
-        runAsyncTest(
-            function () {
-                window.QUnitGlobals.waitForIframe(iframe)
-                    .then(function () {
-                        equal(iframe.contentWindow.location.pathname, '/sessionId!i/https://example.com/test-resource/runner-iframe.html', 'path is correct before click on link');
+        var watchdog = wait(10000);
 
-                        iframe.contentDocument.body.appendChild(link);
-                        link.focus();
+        var runTest = function () {
+            return window.QUnitGlobals
+                .waitForIframe(iframe)
+                .then(function () {
+                    equal(iframe.contentWindow.location.pathname, '/sessionId!i/https://example.com/test-resource/runner-iframe.html', 'path is correct before click on link');
 
-                        //NOTE: we need setTimeout to wait for focus in IE
-                        window.setTimeout(function () {
-                            equal(domUtils.getActiveElement(), iframe);
-                            equal(domUtils.getActiveElement(iframe.contentDocument), link);
+                    iframe.contentDocument.body.appendChild(link);
+                    link.focus();
 
-                            actionsAPI.press('enter');
-                        }, 1000);
-                    });
+                    return wait(browserUtils.isIE ? 3000 : 0)
```

Comment:
we should leave the comment why we need this timeout. And do we really need to wait for 3 seconds (it was 1 sec), it looks too much for waiting for focusing. Did you see the test failed because of this?